plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchTableHandle.java

```diff
@@ -57,13 +62,15 @@ public ElasticsearchTableHandle(
             @JsonProperty("schema") String schema,
             @JsonProperty("index") String index,
             @JsonProperty("constraint") TupleDomain<ColumnHandle> constraint,
+            @JsonProperty("regexes") Map<String, String> regexes,
```

Comment:
Hi, @assaf2
I'm going into this PR recently.

I've already realized part of `Like` expression pushdown for **PostgreSqlConnector**, then I start to try **Function pushdown**. But I find there is only a new **regex map** (which I think is only set for `Like` expression pushdown).

So, does that means if I want to realize **Function pushdown**, I need add another variable like `Optional<String> functionCallExpressions` to deliver function expressions into the QueryBuilder?

Reply:
Functions are pushed down through `io.trino.metadata.Metadata#applyProjection` and through `io.trino.metadata.Metadata#applyFilter`. 
This `regexes` map was built using `applyFilter`'s argument `io.trino.spi.connector.Constraint#getExpression` (see `io.trino.plugin.elasticsearch.ElasticsearchMetadata#applyFilter`) and indeed contains only `LIKE`. Please note that this is `Elasticsearch`'s connector.
Regarding adding `Optional<String> functionCallExpressions ` - this is an implementation detail, but you might need more than just an `Optional<String>`, for example, for keeping the function's arguments (see `io.trino.spi.expression.Call`).

Reply:
Thanks for reply!

> Functions are pushed down through `io.trino.metadata.Metadata#applyProjection` and through `io.trino.metadata.Metadata#applyFilter`. 

Yes, they are for projection expression pushdown and filter expression pushdown. But I'll check other method like `applyJoin` and so on.

I'll update problems I met in implementation in #10155.